### ISSUE TITLE
Undo macros all at once.

### ISSUE THREAD

**Issue by uwumouse** [helix-editor/helix#4448]
When using macros multiplle times (e.g. `20q` with a macro recorded to `@` register) and then pressing `u` to undo, it reverts only last command of the last macro (not even the whole last macro). I think it should be done like it is in `nvim`: `u` keys jumps back to state when all macros are not applyied yet.

**Comment by ksandvik** [comment]
Yes, I would think this should be a unique transaction recorded and then reverted.

**Comment by askreet** [comment]
@Omnikar If you want to push what you have here I'd be willing to get it up to date and shepard it along.

**Comment by robinmoussu** [comment]
Coming from vim, this was a very unexpected behavior. As proposed above, making `q` a unique transaction would be much nicer :)
